### DISCUSSION TITLE
[Core Protocol] Is the SWS what holds a sequencer from stalling or producing L2 blocks pointing to the same L1 origin forever?

### DISCUSSION THREAD

**Discussion by DefiCake** [ethereum-optimism/developers#265] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I am (_still_) going through the documentation and there is one blind spot that I cannot seem to figure out, related to the derivation specification. I am trying to understand what keeps a honest sequencer from going rogue when it comes to including `deposit transactions` in the L2 chain. This is kind of related to other question of mine: https://github.com/ethereum-optimism/developers/discussions/226 

All L2 blocks are tied to a L1 block origin. Blocks posted in DA have time from L1 block origin to 3600 blocks ahead (the SWS, or sequencing window) to be posted.

So, in an scenario in which I have sent a `deposit transaction` at L1 block 100, and the sequencer kept producing L2 blocks  indefinitely pointing to L1 block 99 as origin (or not producing anything at all) until past the SWS, is a hard fork or network reset necessary to get my deposit transaction included?

**Comment by opfocus** [comment]
The derivation is performed by op-node, which reads data from L1. It doesn't necessarily have to be done through the sequencer. 
The sequencer produces blocks, bundling transactions on L2 and deposit transactions (which are mandatory) into blocks.

It's also important to note that users can withdraw their funds forcibly without relying on the sequencer's normal operation.

My response may contain misconceptions, but there's definitely much more to understand.

In addition to the documentation, you can also refer to the relevant content at https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/derivation.md.

**Reply by DefiCake** [reply]
> The derivation is performed by op-node, which reads data from L1

Let's focus on this for a moment. If a faulty sequencer / batcher duo keep posting batches of L2 transactions that point to the same L1 block origin, and those batches end up getting included after the SWS, does that essentially warrant a hard fork of the L2 chain?
